R/flea_starters.R

```diff
@@ -1,70 +1,134 @@
+    dplyr::mutate(
+      position = purrr::map_chr(.data$position, purrr::pluck, "label"),
+      positionColor = NULL
+    ) %>%
+    tidyr::pivot_longer(c("home", "away"), names_to = "franchise", values_to = "player") %>%
+    tidyr::hoist("player", "proPlayer", "owner", "points" = "viewingActualPoints") %>%
+    tidyr::hoist("proPlayer",
+      "player_id" = "id",
+      "player_name" = "nameFull",
+      "pos" = "position",
+      "team" = "proTeamAbbreviation"
+    ) %>%
+    dplyr::filter(!is.na(.data$player_id)) %>%
+    tidyr::hoist("owner", "franchise_id" = "id", "franchise_name" = "name") %>%
+    tidyr::hoist("points", "player_score" = "value") %>%
+    dplyr::select(dplyr::any_of(c(
+      "franchise_id",
+      "franchise_name",
+      "starter_status" = "position",
+      "player_id",
+      "player_name",
+      "pos",
+      "team",
+      "player_score"
+    )))
+
+  return(x)
+}
+
+.flea_starters_detail <- function(week, game_id, conn) {
+  cols <- c(injurytypeAbbreviaition = NA, injurytypeFull = NA, injurydescription = NA, injuryseverity = NA)
+  x <- fleaflicker_getendpoint("FetchLeagueBoxscore",
+                                         sport = "NFL",
+                                         scoring_period = week,
+                                         fantasy_game_id = game_id,
+                                         league_id = conn$league_id
+  ) |>
+    purrr::pluck("content", "lineups") |>
+    list() |>
+    tibble::tibble() |>
+    tidyr::unnest_longer(1) |>
+    tidyr::unnest_wider(1) |>
+    tidyr::unnest_longer("slots") |>
+    tidyr::unnest_wider("slots") |>
+    dplyr::mutate(
+      position = purrr::map_chr(.data$position, purrr::pluck, "label"),
+      positionColor = NULL
+    ) |>
+    tidyr::pivot_longer(c("home", "away"), names_to = "franchise", values_to = "player") |>
+    tidyr::hoist("player", "proPlayer", "owner", "points" = "viewingActualPoints") |>
+    tidyr::hoist("proPlayer",
+                 "player_id" = "id",
+                 "player_name" = "nameFull",
+                 "pos" = "position",
+                 "team" = "proTeamAbbreviation",
+                 "injury" = "injury"
+    ) |>
+    # dplyr::filter(!is.na(.data$player_id)) |>
+    tidyr::unnest_wider("injury", names_sep = "") |>
+    tidyr::hoist("owner", "franchise_id" = "id", "franchise_name" = "name") |>
+    tidyr::hoist("points", "player_score" = "value") |>
+    dplyr::mutate(group = dplyr::case_when(is.na(group) == TRUE ~ "BENCH", .default = group))
+  x = tibble::add_column(x, !!!cols[setdiff(names(cols), names(x))]) |>  
+    dplyr::select(dplyr::any_of(c(
+      "group",
+      "franchise",
+      "franchise_id",
+      "franchise_name",
+      "starter_status" = "position",
+      "player_id",
+      "player_name",
+      "pos",
+      "team",
+      "injurytypeAbbreviaition",
+      "injurytypeFull",
+      "injurydescription",
+      "injuryseverity",
+      "player_score"
+    )))
+  
+  
+  return(x)
+}
```

Comment:
Hmm. I'm not super sure that injuries would belong in ff_starters, but I think it's probably a good spot for this (unless injury status should actually go in ff_rosters for a given week?) . 

If it does end up here, I'd prefer to see this being added to the main ff_starters logic rather than as a separate function.